custom_components/tahoma/binary_sensor.py

```diff
@@ -64,16 +64,22 @@ async def async_setup_entry(hass, entry, async_add_entities):
 class TahomaBinarySensor(TahomaDevice, BinarySensorEntity):
     """Representation of a TaHoma Binary Sensor."""
 
-    def __init__(self, tahoma_device, controller):
-        """Initialize the sensor."""
-        super().__init__(tahoma_device, controller)
-
-        self._state = None
-
     @property
     def is_on(self):
         """Return the state of the sensor."""
-        return self._state == STATE_ON
+
+        states = self.tahoma_device.active_states
+
+        return (
+            states.get(CORE_CONTACT_STATE) == "open"
```

Comment:
Nice one! Great to have the active_states standardized, so we can move to a single update() method in `tahoma_device.py`

Reply:
Yep, it will remove a lot of boilerplate.